### DISCUSSION TITLE
Questions about referencing and using code from ludusavi and ludusavi-manifest

### DISCUSSION THREAD

**Discussion by mcthesw** [mtkennerly/ludusavi#372] [Q&A] [answered]
I'm currently developing a [game save manager](https://github.com/mcthesw/game-save-manager) under the AGPL license, which has a slightly different focus than ludusavi, concentrating more on save/load and game route functionalities for games with limited save management capabilities. I'm looking to implement path parsing and game recognition features and believe ludusavi-manifest could be helpful.

As I'm new to open-source collaboration, I have a few questions:
1. Can I use some static code like placeholder implementations from your project, with proper attribution?
2. Is it okay to reference your code for inspiration while implementing my own solutions?
3. Should I provide attribution and include the MIT license for all instances where I've used or referenced your code? Are there any other steps I should take?

I appreciate any guidance you can provide. If I've said anything inappropriate, I apologize.

Thank you for your time and your excellent project!

**Comment by mtkennerly** [comment, marked as answer]
Hi! Thanks for reaching out :)

> Can I use some static code like placeholder implementations from your project, with proper attribution?

Yes, that's absolutely fine. The way I understand it is that the copied code would still be under the MIT, but the rest of your code would remain AGPL.

> Is it okay to reference your code for inspiration while implementing my own solutions?

Sure! I don't know the legal distinction between copying a "substantial portion" of the code vs using it for inspiration, but for inspiration, I think just adding a comment would be fine (`// Inspired by <link to original code>`).

> Should I provide attribution and include the MIT license for all instances where I've used or referenced your code? Are there any other steps I should take?

Just including the attribution and license is enough.

> I appreciate any guidance you can provide. If I've said anything inappropriate, I apologize.

No worries! This is exactly why I chose a permissive license, so that other projects can also benefit from it.

**Reply by mcthesw** [reply]
Thank you very much for your reply
